### PR TITLE
flake: namespaced checks output

### LINKED ISSUE #152: Namespaced checks output
Working on #125, I'd like to be able to easily list all of the NixOS test checks, because I'd like to build them one by one to see whether each trigger that warning. They are distinguishable in the output of `nix flake show`:

```
│       ├───kbin-frontend: derivation 'kbin-frontend-0.0.1'
│       ├───kbin-toplevel: derivation 'nixos-system-nixos-24.05.20231218.886c9ae'
```

but I'd prefer namespacing, like this, from https://github.com/molybdenumsoftware/pr-tracker/:

```
github:molybdenumsoftware/pr-tracker/d7269959cc6b3f1309c9cf159ad3b89d770d0082
├───checks
│   ├───aarch64-darwin
│   │   ├───clippy omitted (use '--all-systems' to show)
│   │   ├───formatting omitted (use '--all-systems' to show)
│   │   ├───"nixosTests/api" omitted (use '--all-systems' to show)
│   │   ├───"packages/api" omitted (use '--all-systems' to show)
│   │   └───"packages/fetcher" omitted (use '--all-systems' to show)
```

That way they are grouped, and I can trivially see the list of them. I know I'm being picky here, and I can grep my way into a list, but this seems more orderly and and will prevent name conflicts.

### PR DESCRIPTION
**Pull request by mightyiam**
Closes #152

### CODE REVIEW

**Comment by mightyiam** [comment]
I expected a different in `nix flake show`...

**Comment by mightyiam** [comment]
```
git+file:///home/mightyiam/src/ngi-nix/ngipkgs?ref=refs/heads/namespaced-checks&rev=1cc7672968ecfc03c52856ee653e618fc2fa2cce
├───checks
│   └───x86_64-linux
│       ├───formatting: derivation 'treefmt-check'
│       ├───"nixosTests/flarum": derivation 'nixos-system-test-flarum-24.05.20231218.886c9ae'
│       ├───"nixosTests/kbin": derivation 'nixos-system-nixos-24.05.20231218.886c9ae'
│       ├───"nixosTests/pretalx-mysql": derivation 'nixos-system-server-24.05.20231218.886c9ae'
│       ├───"nixosTests/pretalx-postgresql": derivation 'nixos-system-server-24.05.20231218.886c9ae'
│       ├───"packages/anastasis": derivation 'anastasis-0.4.0'
│       ├───"packages/atomic-cli": derivation 'atomic-cli-0.34.5'
│       ├───"packages/atomic-server": derivation 'atomic-server-0.34.5'
│       ├───"packages/autobase": derivation 'autobase'
│       ├───"packages/corestore": derivation 'corestore-6.15.9'
│       ├───"packages/euclid3": derivation 'python3.11-euclid3-0.01'
│       ├───"packages/flarum": derivation 'flarum-1.8.0'
│       ├───"packages/freeDiameter": derivation 'freeDiameter-1.5.0'
│       ├───"packages/gnunet-messenger-cli": derivation 'gnunet-messenger-cli-0.1.1'
│       ├───"packages/hyperbeam": derivation 'hyperbeam-3.0.1'
│       ├───"packages/hyperblobs": derivation 'hyperblobs-2.3.3'
│       ├───"packages/hypercore": derivation 'hypercore-10.28.11'
│       ├───"packages/hyperswarm": derivation 'hyperswarm-4.7.3'
│       ├───"packages/kbin": derivation 'kbin'
│       ├───"packages/kbin-backend": derivation 'kbin-0.0.1'
│       ├───"packages/kbin-frontend": derivation 'kbin-frontend-0.0.1'
│       ├───"packages/kikit": derivation 'python3.11-kikit-1.3.0'
│       ├───"packages/kip": derivation 'kip-unstable-2021-07-27'
│       ├───"packages/lcrq": derivation 'lcrq-0.1.0'
│       ├───"packages/lcsync": derivation 'lcsync-0.2.1'
│       ├───"packages/leaf": derivation 'leaf-unstable-2020-04-28'
│       ├───"packages/libgnunetchat": derivation 'libgnunetchat'
│       ├───"packages/librecast": derivation 'librecast-0.7-RC3'
│       ├───"packages/lillydap": derivation 'lillydap-0.9.2'
│       ├───"packages/mcaptcha": derivation 'mcaptcha-unstable-2023-07-04'
│       ├───"packages/mcaptcha-cache": derivation 'cache-unstable-2023-03-08'
│       ├───"packages/pcbnew-transition": derivation 'python3.11-pcbnewTransition-0.3.4'
│       ├───"packages/pretalx": derivation 'pretalx-2023.1.3'
│       ├───"packages/pretalx-downstream": derivation 'python3.11-pretalx-downstream-1.1.5'
│       ├───"packages/pretalx-frontend": derivation 'pretalx-frontend-2023.1.0'
│       ├───"packages/pretalx-full": derivation 'pretalx-2023.1.3'
│       ├───"packages/pretalx-media-ccc-de": derivation 'python3.11-pretalx-media-ccc-de-1.1.1'
│       ├───"packages/pretalx-pages": derivation 'python3.11-pretalx-pages-1.3.3'
│       ├───"packages/pretalx-public-voting": derivation 'python3.11-pretalx-public-voting-1.3.0'
│       ├───"packages/pretalx-venueless": derivation 'python3.11-pretalx-venueless-1.3.0'
│       ├───"packages/pybars3": derivation 'python3.11-pybars3-0.9.7'
│       ├───"packages/pymeta3": derivation 'python3.11-pymeta3-0.5.1'
│       ├───"packages/quicksasl": derivation 'quicksasl-0.11.0'
│       ├───"packages/rosenpass": derivation 'rosenpass-unstable-2023-09-28'
│       ├───"packages/rosenpass-tools": derivation 'rosenpass-tools-unstable-2023-09-28'
│       ├───"packages/steamworks": derivation 'steamworks-0.97.2'
│       ├───"packages/steamworks-pulleyback": derivation 'steamworks-pulleyback-0.3.0'
│       ├───"packages/sync": derivation 'sync-0.9.3'
│       ├───"packages/taler-mdb": derivation 'taler-mdb-0.9.3'
│       ├───"packages/tlspool": derivation 'tlspool-0.9.6'
│       ├───"packages/tlspool-gui": derivation 'tlspool-gui-0.0.6'
│       ├───"packages/twister": derivation 'twister-0.9.3'
│       └───"packages/wireguard-rs": derivation 'wireguard-rs-unstable-2021-01-13'
├───formatter
│   └───x86_64-linux: package 'treefmt'
├───hydraJobs
│   ├───nixosConfigurations
│   │   └───x86_64-linux
│   │       ├───flarum: derivation 'nixos-system-test-flarum-24.05.20231218.886c9ae'
│   │       ├───kbin: derivation 'nixos-system-nixos-24.05.20231218.886c9ae'
│   │       ├───pretalx-mysql: derivation 'nixos-system-server-24.05.20231218.886c9ae'
│   │       └───pretalx-postgresql: derivation 'nixos-system-server-24.05.20231218.886c9ae'
│   ├───packages
│   │   └───x86_64-linux
│   │       ├───anastasis: derivation 'anastasis-0.4.0'
│   │       ├───atomic-cli: derivation 'atomic-cli-0.34.5'
│   │       ├───atomic-server: derivation 'atomic-server-0.34.5'
│   │       ├───autobase: derivation 'autobase'
│   │       ├───corestore: derivation 'corestore-6.15.9'
│   │       ├───euclid3: derivation 'python3.11-euclid3-0.01'
│   │       ├───flarum: derivation 'flarum-1.8.0'
│   │       ├───freeDiameter: derivation 'freeDiameter-1.5.0'
│   │       ├───gnunet-messenger-cli: derivation 'gnunet-messenger-cli-0.1.1'
│   │       ├───hyperbeam: derivation 'hyperbeam-3.0.1'
│   │       ├───hyperblobs: derivation 'hyperblobs-2.3.3'
│   │       ├───hypercore: derivation 'hypercore-10.28.11'
│   │       ├───hyperswarm: derivation 'hyperswarm-4.7.3'
│   │       ├───kbin: derivation 'kbin'
│   │       ├───kbin-backend: derivation 'kbin-0.0.1'
│   │       ├───kbin-frontend: derivation 'kbin-frontend-0.0.1'
│   │       ├───kikit: derivation 'python3.11-kikit-1.3.0'
│   │       ├───kip: derivation 'kip-unstable-2021-07-27'
│   │       ├───lcrq: derivation 'lcrq-0.1.0'
│   │       ├───lcsync: derivation 'lcsync-0.2.1'
│   │       ├───leaf: derivation 'leaf-unstable-2020-04-28'
│   │       ├───libgnunetchat: derivation 'libgnunetchat'
│   │       ├───librecast: derivation 'librecast-0.7-RC3'
│   │       ├───lillydap: derivation 'lillydap-0.9.2'
│   │       ├───mcaptcha: derivation 'mcaptcha-unstable-2023-07-04'
│   │       ├───mcaptcha-cache: derivation 'cache-unstable-2023-03-08'
│   │       ├───pcbnew-transition: derivation 'python3.11-pcbnewTransition-0.3.4'
│   │       ├───pretalx: derivation 'pretalx-2023.1.3'
│   │       ├───pretalx-downstream: derivation 'python3.11-pretalx-downstream-1.1.5'
│   │       ├───pretalx-frontend: derivation 'pretalx-frontend-2023.1.0'
│   │       ├───pretalx-full: derivation 'pretalx-2023.1.3'
│   │       ├───pretalx-media-ccc-de: derivation 'python3.11-pretalx-media-ccc-de-1.1.1'
│   │       ├───pretalx-pages: derivation 'python3.11-pretalx-pages-1.3.3'
│   │       ├───pretalx-public-voting: derivation 'python3.11-pretalx-public-voting-1.3.0'
│   │       ├───pretalx-venueless: derivation 'python3.11-pretalx-venueless-1.3.0'
│   │       ├───pybars3: derivation 'python3.11-pybars3-0.9.7'
│   │       ├───pymeta3: derivation 'python3.11-pymeta3-0.5.1'
│   │       ├───quicksasl: derivation 'quicksasl-0.11.0'
│   │       ├───rosenpass: derivation 'rosenpass-unstable-2023-09-28'
│   │       ├───rosenpass-tools: derivation 'rosenpass-tools-unstable-2023-09-28'
│   │       ├───steamworks: derivation 'steamworks-0.97.2'
│   │       ├───steamworks-pulleyback: derivation 'steamworks-pulleyback-0.3.0'
│   │       ├───sync: derivation 'sync-0.9.3'
│   │       ├───taler-mdb: derivation 'taler-mdb-0.9.3'
│   │       ├───tlspool: derivation 'tlspool-0.9.6'
│   │       ├───tlspool-gui: derivation 'tlspool-gui-0.0.6'
│   │       ├───twister: derivation 'twister-0.9.3'
│   │       └───wireguard-rs: derivation 'wireguard-rs-unstable-2021-01-13'
│   └───tests
│       └───x86_64-linux
│           ├───kbin
│           │   └───kbin: derivation 'vm-test-run-kbin'
│           ├───libgnunetchat
│           │   └───pkg-config: derivation 'check-meta-pkg-config-modules-for-libgnunetchat'
│           ├───mcaptcha
│           │   ├───mcaptcha-bring-your-own-services: derivation 'vm-test-run-test-of-mcaptcha-with-database-and-other-services-running-on-a-different-node'
│           │   └───mcaptcha-create-locally: derivation 'vm-test-run-test-of-mcaptcha-with-database-and-other-services-all-running-on-the-same-node'
│           ├───pretalx
│           │   └───pretalx: derivation 'vm-test-run-pretalx-tests'
│           ├───pretalx-full
│           │   └───pretalx: derivation 'vm-test-run-pretalx-tests'
│           └───rosenpass
│               └───rosenpass: derivation 'vm-test-run-rosenpass'
├───nixosConfigurations
│   ├───flarum: NixOS configuration
│   ├───kbin: NixOS configuration
│   ├───pretalx-mysql: NixOS configuration
│   └───pretalx-postgresql: NixOS configuration
├───nixosModules
│   ├───default: NixOS module
│   ├───flarum: NixOS module
│   ├───kbin: NixOS module
│   ├───mcaptcha: NixOS module
│   ├───pretalx: NixOS module
│   └───unbootable: NixOS module
├───overlays
│   └───default: Nixpkgs overlay
└───packages
    └───x86_64-linux
        ├───anastasis: package 'anastasis-0.4.0'
        ├───atomic-cli: package 'atomic-cli-0.34.5'
        ├───atomic-server: package 'atomic-server-0.34.5'
        ├───autobase: package 'autobase'
        ├───corestore: package 'corestore-6.15.9'
        ├───euclid3: package 'python3.11-euclid3-0.01'
        ├───flarum: package 'flarum-1.8.0'
        ├───freeDiameter: package 'freeDiameter-1.5.0'
        ├───gnunet-messenger-cli: package 'gnunet-messenger-cli-0.1.1'
        ├───hyperbeam: package 'hyperbeam-3.0.1'
        ├───hyperblobs: package 'hyperblobs-2.3.3'
        ├───hypercore: package 'hypercore-10.28.11'
        ├───hyperswarm: package 'hyperswarm-4.7.3'
        ├───kbin: package 'kbin'
        ├───kbin-backend: package 'kbin-0.0.1'
        ├───kbin-frontend: package 'kbin-frontend-0.0.1'
        ├───kikit: package 'python3.11-kikit-1.3.0'
        ├───kip: package 'kip-unstable-2021-07-27'
        ├───lcrq: package 'lcrq-0.1.0'
        ├───lcsync: package 'lcsync-0.2.1'
        ├───leaf: package 'leaf-unstable-2020-04-28'
        ├───libgnunetchat: package 'libgnunetchat'
        ├───librecast: package 'librecast-0.7-RC3'
        ├───lillydap: package 'lillydap-0.9.2'
        ├───mcaptcha: package 'mcaptcha-unstable-2023-07-04'
        ├───mcaptcha-cache: package 'cache-unstable-2023-03-08'
        ├───pcbnew-transition: package 'python3.11-pcbnewTransition-0.3.4'
        ├───pretalx: package 'pretalx-2023.1.3'
        ├───pretalx-downstream: package 'python3.11-pretalx-downstream-1.1.5'
        ├───pretalx-frontend: package 'pretalx-frontend-2023.1.0'
        ├───pretalx-full: package 'pretalx-2023.1.3'
        ├───pretalx-media-ccc-de: package 'python3.11-pretalx-media-ccc-de-1.1.1'
        ├───pretalx-pages: package 'python3.11-pretalx-pages-1.3.3'
        ├───pretalx-public-voting: package 'python3.11-pretalx-public-voting-1.3.0'
        ├───pretalx-venueless: package 'python3.11-pretalx-venueless-1.3.0'
        ├───pybars3: package 'python3.11-pybars3-0.9.7'
        ├───pymeta3: package 'python3.11-pymeta3-0.5.1'
        ├───quicksasl: package 'quicksasl-0.11.0'
        ├───rosenpass: package 'rosenpass-unstable-2023-09-28'
        ├───rosenpass-tools: package 'rosenpass-tools-unstable-2023-09-28'
        ├───steamworks: package 'steamworks-0.97.2'
        ├───steamworks-pulleyback: package 'steamworks-pulleyback-0.3.0'
        ├───sync: package 'sync-0.9.3'
        ├───taler-mdb: package 'taler-mdb-0.9.3'
        ├───tlspool: package 'tlspool-0.9.6'
        ├───tlspool-gui: package 'tlspool-gui-0.0.6'
        ├───twister: package 'twister-0.9.3'
        └───wireguard-rs: package 'wireguard-rs-unstable-2021-01-13'
```

File: flake.nix [hunk 1]
```diff
@@ -107,7 +107,7 @@
     eachDefaultSystemOutputs = flake-utils.lib.eachDefaultSystem (system: let
       pkgs = importNixpkgs system [rust-overlay.overlays.default];
       treefmtEval = loadTreefmt pkgs;
-      toplevel = name: config: nameValuePair "${name}-toplevel" config.config.system.build.toplevel;
+      toplevel = name: config: nameValuePair "nixosTests/${name}" config.config.system.build.toplevel;
```
Comment (mightyiam) [inline flake.nix:110]: Is this `-toplevel` suffix a useful indication of anything?

Comment (lorenzleutgeb) [reply flake.nix]: Yes. That it builds `config.system.build.toplevel` of some `config`, which has the name `name`. The derivation is *not* a NixOS test, as the new name suggests. I am strongly against this change.

**Review by lorenzleutgeb — APPROVED** [review]

**Pull request merged**